conf/controller/scoring.sql

```diff
@@ -28,49 +28,121 @@ WITH
     FROM flagdefense
     GROUP BY team_id, service_id
   ),
-  sla_ok AS (
-    SELECT count(*) as sla_ok,
-           team_id,
-           service_id
-    FROM scoring_statuscheck
-    WHERE status = 0
-    GROUP BY team_id, service_id
-  ),
-  sla_recover AS (
-    SELECT 0.5 * count(*) as sla_recover,
-           team_id,
-           service_id
-    FROM scoring_statuscheck
-    WHERE status = 4
-    GROUP BY team_id, service_id
-  ),
+--  sla_ok AS (
+--    SELECT count(*) as sla_ok,
+--           team_id,
+--           service_id
+--    FROM scoring_statuscheck
+--    WHERE status = 0
+--    GROUP BY team_id, service_id
+--  ),
+--  sla_recover AS (
+--    SELECT 0.5 * count(*) as sla_recover,
+--           team_id,
+--           service_id
+--    FROM scoring_statuscheck
+--    WHERE status = 4
+--    GROUP BY team_id, service_id
+--  ),
   teams as (
     SELECT user_id as team_id
     FROM registration_team
     INNER JOIN auth_user ON auth_user.id = registration_team.user_id
     WHERE is_active = true
 	  AND nop_team = false
   ),
-  sla AS (
+--  sla AS (
+--    SELECT (SELECT sqrt(count(*)) FROM teams) * (coalesce(sla_ok, 0) + coalesce(sla_recover, 0)) as sla,
+--           team_id,
+--           service_id
+--    FROM sla_ok
+--    NATURAL FULL OUTER JOIN sla_recover
+--  ),
+  fill AS (
+    SELECT team_id, scoring_servicegroup.id AS service_group_id
+    FROM teams, scoring_servicegroup
+  ),
+  servicegroup AS (
+    SELECT scoring_service.service_group_id AS service_group_id,
+           count(scoring_service.id) AS services_count
+    FROM scoring_service
+    GROUP BY scoring_service.service_group_id
+  ),
+  attack_by_servicegroup AS (
+    SELECT team_id,
+           scoring_service.service_group_id AS service_group_id,
+           sum(attack) AS attack,
+           sum(bonus) AS bonus
+    FROM attack
+    INNER JOIN scoring_service ON scoring_service.id = attack.service_id
+    GROUP BY team_id, scoring_service.service_group_id
+  ),
+  defense_by_servicegroup AS (
+    SELECT team_id,
+           scoring_service.service_group_id AS service_group_id,
+           sum(defense) AS defense
+    FROM defense
+    INNER JOIN scoring_service ON scoring_service.id = defense.service_id
+    GROUP BY team_id, scoring_service.service_group_id
+  ),
+  sla_ok_by_servicegroup_tick AS (
+    SELECT team_id,
+           scoring_service.service_group_id AS service_group_id,
+           0.5 * servicegroup.services_count * (COUNT(*) = servicegroup.services_count)::int AS sla_ok
```

Comment:
This is the SLA formula for each service group, which should be easy to replace in case we want to use another formula.

Reply:
This actually evaluates to `0.5 * len(services[group])` for each check result instead of `0.5 * len(services[group])**2` as I initially thought.

But, because it gets evaluated once for each service in the group this still results in `0.5 * len(services[group])**2`. I think this was unintended, as their [scoring](https://docs.ecsc2022.eu/scoring/) provides the formula
```
sla = (count(ticks_with_status['up'] + 0.5 * ticks_with_status['recovering'])) * sqrt(count(teams)) * count(flag_stores_in_service)
```
Also note that this formula is the same as the formula for recovering sla points, so this formula should have had a factor of `1.0` instead.

To get the documented behavior, the formulas should have been:
sla: `1.0 * (COUNT(*) = servicegroup.services_count)::int`
sla_recover: `0.5 * (COUNT(*) = servicegroup.services_count)::int`

Some additional thoughts:
* Do we want sla to scale linearly with the number of flagstores? Other options include sla that doesn't depend on the number of flagstores  or some non-linear scaling (maybe `sqrt(len(services[group])`?)
* Currently no partial sla points are awarded for having some flagstores available, when others are down. Maybe we should award `y/x * sla`  if y of x services in the group are up?